pkg/kmm/preflightvalidationocp_test.go

```diff
@@ -0,0 +1,416 @@
+
+	testBuilder.WithOptions(func(builder *PreflightValidationOCPBuilder) (*PreflightValidationOCPBuilder, error) {
+		return builder, nil
+	})
+
+	assert.Equal(t, "", testBuilder.errorMsg)
+
+	testBuilder.WithOptions(func(builder *PreflightValidationOCPBuilder) (*PreflightValidationOCPBuilder, error) {
+		return builder, fmt.Errorf("error")
+	})
+
+	assert.Equal(t, "error", testBuilder.errorMsg)
+}
+
+func TestPullPreflightValidationOCP(t *testing.T) {
+	testCases := []struct {
+		name                string
+		namespace           string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		client              bool
+	}{
+		{
+			name:                "test",
+			namespace:           "testns",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidation 'apiClient' cannot be empty",
+			client:              false,
+		},
+		{
+			name:                "test",
+			namespace:           "testns",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "",
+			client:              true,
+		},
+		{
+			name:                "",
+			namespace:           "testns",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidationocp 'name' cannot be empty",
+			client:              true,
+		},
+		{
+			name:                "test",
+			namespace:           "",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidationocp 'nsname' cannot be empty",
+			client:              true,
+		},
+		{
+			name:                "test",
+			namespace:           "testns",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidationocp object test doesn't exist in namespace testns",
+			client:              true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testPreflight := generatePreflight(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testPreflight)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		builderResult, err := PullPreflightValidationOCP(testSettings, testCase.name, testCase.namespace)
+
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.NotNil(t, builderResult)
+			assert.Equal(t, testCase.name, builderResult.Definition.Name)
+			assert.Equal(t, testCase.namespace, builderResult.Definition.Namespace)
+		}
+	}
+}
+
+func TestPreflightCreate(t *testing.T) {
+	testCases := []struct {
+		testPreflight *PreflightValidationOCPBuilder
+		expectedError error
+	}{
+		{
```

Comment:
Please add test case when object already exist on cluster.

Reply:
Updated the test and refactored the ResourceVersion part